packages/make-styles/src/runtime/resolveStyleRules.ts

```diff
@@ -17,20 +17,39 @@ import { hashClassName } from './utils/hashClassName';
 import { resolveProxyValues } from './createCSSVariablesProxy';
 import { hashPropertyKey } from './utils/hashPropertyKey';
 
-/**
- * Transforms input styles to resolved rules: generates classnames and CSS.
- *
- * @internal
- */
-export function resolveStyleRules(
+function pushToClassesMap(
+  classesMap: CSSClassesMap,
+  propertyKey: string,
+  ltrClassname: string,
+  rtlClassname: string | undefined,
+) {
+  classesMap[propertyKey] = rtlClassname ? [ltrClassname!, rtlClassname] : ltrClassname;
+}
+
+function pushToCSSRules(
+  cssRulesByBucket: CSSRulesByBucket,
+  styleBucketName: StyleBucketName,
+  ltrCSS: string,
+  rtlCSS: string | undefined,
+) {
+  cssRulesByBucket[styleBucketName] = cssRulesByBucket[styleBucketName] || [];
+  cssRulesByBucket[styleBucketName]!.push(ltrCSS);
+
+  if (rtlCSS) {
+    cssRulesByBucket[styleBucketName]!.push(rtlCSS);
+  }
+}
+
+function resolveStyleRulesInner(
   styles: MakeStyles,
   unstable_cssPriority: number = 0,
   pseudo = '',
   media = '',
   support = '',
-  result: Record<string, MakeStylesResolvedRule> = {},
+  cssClassesMap: CSSClassesMap = {},
+  cssRulesByBucket: CSSRulesByBucket = {},
   rtlValue?: string,
-): Record<string, MakeStylesResolvedRule> {
+): [CSSClassesMap, CSSRulesByBucket] {
   const expandedStyles: MakeStyles = expandShorthand(resolveProxyValues(styles));
```

Comment:
This must be removed here as it is now responsibility of `resolveStyleRules`

Reply:
Oops 🤦‍♂️ Fixed in 2cabe2d